src/segger/data/io.py

```diff
@@ -963,21 +1009,24 @@ def build_pyg_data_from_tile(
         transcripts_df["token"] = token_encoding  # Store the integer tokens in the 'token' column
         data["tx"].token = torch.as_tensor(token_encoding).int()
         # Handle additional embeddings lazily as well
-        if not self.embedding_df.empty:
+        if self.embedding_df is not None and not self.embedding_df.empty:
             embeddings = delayed(lambda df: self.embedding_df.loc[df[self.keys.FEATURE_NAME.value].values].values)(
                 transcripts_df
             )
         else:
             embeddings = token_encoding
-        embeddings = embeddings.compute()
+        if hasattr(embeddings, "compute"):
+            embeddings = embeddings.compute()
         x_features = torch.as_tensor(embeddings).int()
         data["tx"].x = x_features
 
         # Check if the overlap column exists, if not, compute it lazily using Dask
         if self.keys.OVERLAPS_BOUNDARY.value not in transcripts_df.columns:
             if self.verbose:
                 print(f"Computing overlaps for transcripts...")
-            transcripts_df = self.compute_transcript_overlap_with_boundaries(transcripts_df, bd_gdf, scale_factor=1.0)
+            transcripts_df = self.compute_transcript_overlap_with_boundaries(
```

Comment:
`bd_gdf` is a `GeoDataFrame` but it was passed as a positional argument, so as if it was a (Dask?) dataframe.